### DISCUSSION TITLE
Icon color

### DISCUSSION THREAD

**Discussion by madcoda9000** [Megabit/Blazorise#4129] [Q&A] [answered]
Hello,

I've st up an own color scheme in app.razor following the documentation. Which works as expected.

Except the icons :-) When i define an icon like this:  `<BarIcon IconName="IconName.Dashboard" TextColor="TextColor.Primary" />`

It becomes the default bootstrap primary color and not my theme primary color. How can i change this? It seems that there is no color attribute available.

The theme is defined as follows:

```
<Blazorise.ThemeProvider Theme="@mtheme">
    <Router AppAssembly="@typeof(Program).Assembly">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)">
                <Authorizing>
                    Authorizing...
                </Authorizing>
                <NotAuthorized>
                    <RedirectToLogin />
                </NotAuthorized>
            </AuthorizeRouteView>
        </Found>
        <NotFound>
            <CascadingAuthenticationState>
                <LayoutView Layout="@typeof(MainLayout)">
                    <p>Sorry, there's nothing at this address.</p>
                </LayoutView>
            </CascadingAuthenticationState>
        </NotFound>
    </Router>
</Blazorise.ThemeProvider>

@code{
    private Theme mtheme = new Theme
    {
        ColorOptions = new ThemeColorOptions
        {
            Primary = "#090251",
            Secondary = "#f5023c",
            Danger = "#f44336ff",
            Warning = "#ff9800ff",
            Info = "#2196f3ff",
            Success = "#00c853ff",
            Link = "#f5023c",
            Light = "#ffffff"
        }
    };
}
```

**Comment by stsrki** [comment, marked as answer]
You also need to define `TextColorOptions` 

https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/App.razor#L48

**Reply by madcoda9000** [reply]
This is awsome! Thank you. The only thing left is that if i define a link like that `<Link To="/Auth/register">register..</Link>`

the defined color from my theme is not applied too.

**Reply by madcoda9000** [reply]
ok, found it by myself. i've to manually aadd the textcolor attribute to the link. But i wonder what the link color in theme settings is for if it is not applied to all links automatically.

**Comment by stsrki** [comment]
The `ThemeColorOptions.Link` was meant to control something that has a CSS class with a `-link` name, for example, `btn-link`. The problem with the `Link` component is that it doesn't have any predefined styles as the `Button` component has.

Although it kinda makes sense to have it controlled by `ThemeColorOptions.Link`. 🤔I will need to think this through...

**Reply by madcoda9000** [reply]
ok, i understand. Many thx
